### DISCUSSION TITLE
Automating my henhouse door with a ESP12F_Relay_X8

### DISCUSSION THREAD

**Discussion by frankhommers** [arendst/Tasmota#15494] [Q&A] [answered]
I have flashed Tasmota to this board: https://templates.blakadder.com/ESP12F_Relay_X8.html

I've hooked up a linear actuator to this board. The linear actuator expands when I send this command in the console:
`Backlog Power5 On; Power6 On; Power7 Off; Power8 Off; Delay 600; Power5 Off; Power6 Off; Power7 Off; Power8 Off`
And it retracts like so:
`Backlog Power5 On; Power6 On; Power7 On; Power8 On; Delay 600; Power5 Off; Power6 Off; Power7 Off; Power8 Off`

What I would like to do now (in order of priority ;-) )

1) Remove relays 5-8 from the WebUI and replace them with a button: Toggle door (And I would like to see the current state above it).
2) Trigger the open/close action by MQTT.
3) Being able to remember state after reboots.

I this possible? And can someone guide me a bit? I am kind of stuck on it ;-)

**Comment by barbudor** [comment]
Hi
You should probably start here: https://tasmota.github.io/docs/Blinds-and-Shutters/

Customization of web UI is limited however using shutter modes will change some buttons to open/close that should be close to your objective

**Reply by frankhommers** [reply]
That sounds kind of hard. I'll look into it ;-) Still hoping for somewhat simpler solution ;-)

**Reply by frankhommers** [reply]
I have tried it, but the interlocking does not what I need.

**Reply by barbudor** [reply]
If you forget about web GUI customization and focus only on mqtt, you can create event-rules to trigger you commands above from an event command through mqtt

**Reply by frankhommers** [reply]
@barbudor : Yeah that was what I was thinking, but I couldn't find docs about sending a rule trigger by MQTT.... Do you know where to look?

**Reply by barbudor** [reply]
`event` is a Tasmota command like any other : https://tasmota.github.io/docs/Commands/#event
You send an event through MQTT by publishing on `cmnd/yourtopic/event` and payload is `eventname` or `eventname=value`
Then you create a rule that can capture the event by `ON event#eventname DO anything with %value% ENDON`

In your case that could become:
```
Rule1
   ON event#expand DO Backlog Power5 On; Power6 On; Power7 Off; Power8 Off; Delay 600; Power5 Off; Power6 Off; Power7 Off; Power8 Off ENDON
   ON event#retract DO Backlog Power5 On; Power6 On; Power7 On; Power8 On; Delay 600; Power5 Off; Power6 Off; Power7 Off; Power8 Off ENDON
Rule1 1
```
So you just have to publish `extract` or `expand` to `cmnd/yourtopic/event`
However using Delay is not very reliable, especially long delays so you should look into using Rules Timer instead

**Reply by frankhommers** [reply]
Thanks!

**Comment by frankhommers** [comment, marked as answer]
I am going for this solution https://github.com/arendst/Tasmota/discussions/15494#discussioncomment-2672596
